core/server/worker/src/main/java/alluxio/worker/block/MonoBlockStore.java

```diff
@@ -171,7 +172,10 @@ public BlockReader createUfsBlockReader(long sessionId, long blockId, long offse
     try {
       BlockReader reader = mUnderFileSystemBlockStore.createBlockReader(sessionId, blockId, offset,
           positionShort, options);
-      return new DelegatingBlockReader(reader, () -> closeUfsBlock(sessionId, blockId));
+      BlockReader blockReader = new DelegatingBlockReader(reader,
+          () -> closeUfsBlock(sessionId, blockId));
+      Metrics.WORKER_ACTIVE_CLIENTS.inc();
```

Comment:
I think we add client when create blockReader?
```
    if (blockMeta.isPresent()) {
      reader = mLocalBlockStore.createBlockReader(sessionId, blockId, offset);
    } else {
      boolean checkUfs = options != null && (options.hasUfsPath() || options.getBlockInUfsTier());
      if (!checkUfs) {
        throw new BlockDoesNotExistRuntimeException(blockId);
      }
      // When the block does not exist in Alluxio but exists in UFS, try to open the UFS block.
      reader = createUfsBlockReader(sessionId, blockId, offset, positionShort, options);
    }
    DefaultBlockWorker.Metrics.WORKER_ACTIVE_CLIENTS.inc();
```

Reply:
private boolean cacheBlockFromUfs(long blockId, long blockSize,
      Protocol.OpenUfsBlockOptions openUfsBlockOptions) throws IOException {
    try (BlockReader reader = mBlockWorker.createUfsBlockReader(
        Sessions.CACHE_UFS_SESSION_ID, blockId, 0, false, openUfsBlockOptions)) {
      // Read the entire block, caching to block store will be handled internally in UFS block store
      // when close the reader.
      // Note that, we read from UFS with a smaller buffer to avoid high pressure on heap
      // memory when concurrent async requests are received and thus trigger GC.
      long offset = 0;
      while (offset < blockSize) {
        long bufferSize = Math.min(8L * Constants.MB, blockSize - offset);
        reader.read(offset, bufferSize);
        offset += bufferSize;
      }
    }
    return true;
  }

Reply:
But if you remove the metrics here we would have wrong metrics when reading directly from the client...

Reply:
Reading directly from the client-------What does it mean? I don't quite understand.
In terms of function createBlockReader semantics, I think there is no problem. And I added inc in createUfsBlockReader.

Reply:
I see then LGTM